### ISSUE TITLE
alacritty migrate: Configuration file migration failed: parsing error: Unable to locate config file

### ISSUE THREAD

**Issue by Randoragon** [alacritty/alacritty#7473]
Fresh after upgrading to alacritty 0.13.0, I saw in the release notes that there's been a migration from yaml to toml. I ran `alacritty migrate` as instructed, but it failed with:
```
Configuration file migration failed:
    "/home/user/.config/alacritty/alacritty.yml": parsing error: Unable to locate config file
```
Obviously, the file is there. There is a parsing error, but the exact cause seems to be misreported.

After playing with it a bit, I found that running `alacritty migrate -i` (skipping imports) succeeds. The only import in my yaml was `colorscheme.yml`. To be honest, I don't remember what exactly that is, but I've decided to create this issue, because `Unable to locate config file` is misleading and maybe this will help other people.

It's unfortunate that `migrate` fails with a vague error on a [former valid config](https://github.com/Randoragon/dotfiles/blob/b45d7534595d6e3cc6550d616294f8e5bb4c49a4/alacritty/.config/alacritty/alacritty.yml). I don't mean this to come off as picky though, thanks for taking the effort of making this migration smooth!

### System

OS: Arch Linux, X11, bspwm, picom
Version: alacritty 0.13.0 (78fa4d6f)

**Comment by kchibisov** [comment]
yeah, it should likely tell which file it failed to locate correctly.

**Comment by chrisduerr** [comment]
> It's unfortunate that migrate fails with a vague error on a former valid config.

Just to be 100% clear, this is still a valid configuration file and there's no intention on changing that. Missing imports is an intentionally supported feature. Seems like migration just doesn't account for that, it should be a warning instead.

**Comment by mo-st** [comment]
Thanks for posting. I imagine quite a few others ran into this problem, like myself. The correct solution was to first migrate the `coloscheme.yml` files and afterwards rename the imports in the `alacritty.yml` before migrating. 
Maybe this could be done automatically or the user could be prompted to do so in the warning message of the migrate command.

**Comment by aloispichler** [comment]
Same error here. I am on Arch and my config is in ~/.config/alacritty/alacritty.yml

**Comment by aloispichler** [comment]
I could migrate manually. What do the following look like after migration?  
```
key_bindings:
  - { key: Return, mods: Control,       chars: "\e[13;5u" }
  - { key: Return, mods: Shift,         chars: "\e[13;2u" }
  - { key: Tab, mods: Control,          chars: "\e[9;5u" }
  - { key: Tab, mods: Control|Shift,    chars: "\e[9;6u" }
  - { key: Space, mods: Shift,          chars: "\e[32;2u" }
  # - { key: Space, mods: Control,        chars: "\e[64;5u" }
  - { key: Space, mods: Control|Shift,  chars: "\e[32;6u" }
  #- { key: Paste,                                       action: Paste          }
```

**Comment by Randoragon** [comment]
> I could migrate manually. What do the following look like after migration?

Comments are not preserved by the parser, for obvious reasons. Each keybinding is converted to a toml entry like so:
```toml
[[keyboard.bindings]]
chars = "\u001B[13;5u"
key = "Return"
mods = "Control"
```

You can very easily verify this yourself by saving the snippet you've posted into a `file.yml`, and running `alacritty migrate -c file.yml`, which should create `file.toml`. The `-c` option is documented in `alacritty migrate --help`.

Mind that the problem described in this issue is strictly tied to importing non-existent files. This appears to have been solved, a fix should be available in 0.13.1 (whenever that is released), and until then it's very easy to avoid the problem manually by passing the `-i` flag as I described, or getting rid of the redundant import entries from your config, or using the development branch where the fix is present. It's not really clear from your explanation if our problems are at all related.

**Comment by savoygrizzly** [comment]
This issue just happened to me. I had to manually edit the file (which was located at the correct path despite `alacritty migrate` supposedly failing to locate it).
